File: 1097A_Card_Game.cpp
/*
Entrada:
 - Primera línea: carta sobre la mesa (dos caracteres, e.g. "AH")
 - Segunda línea: cinco cartas separadas por espacios
Salida:
 - "YES" si puedes jugar alguna carta (mismo valor o mismo palo), sino "NO"
Ejemplo:
Entrada:
AH
3D 4C 5H 6S 7D
Salida:
YES
*/

#include <iostream>
using namespace std;

int main() {
    string mesa;
    string mano[5];
    cin >> mesa;
    for (int i = 0; i < 5; ++i) {
        cin >> mano[i];
    }

    bool canPlay = false;
    for (int i = 0; i < 5; ++i) {
        if (mesa[0] == mano[i][0] || mesa[1] == mano[i][1]) {
            canPlay = true;
            break;
        }
    }

    if (canPlay)
        cout << "YES" << endl;
    else
        cout << "NO" << endl;

    return 0;
}

File: 1097A_Card_Game.py

"""
Entrada:
 - Primera línea: carta sobre la mesa (dos caracteres, valor+palo, e.g. "AH")
 - Segunda línea: cinco cartas en la mano separadas por espacios (cada una valor+palo)
Salida:
 - "YES" si puedes jugar alguna carta (mismo valor o mismo palo), sino "NO"
Ejemplo:
Entrada:
AH
3D 4C 5H 6S 7D
Salida:
YES
"""

switch = 0
mesa = input()
mano = list(input().split())

for i in range(5):
    if mesa[0] == mano[i][0]:
        print("YES")
        switch = 1
        break
    elif mesa[1] == mano[i][1]:
        print("YES")
        switch = 1
        break

if switch == 0:
    print("NO")